x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowNodeTask.java

```diff
@@ -274,6 +277,13 @@ private void sendShardChangesRequest(long from, int maxOperationCount, long maxR
                         failedReadRequests++;
                         fetchExceptions.put(from, Tuple.tuple(retryCounter, ExceptionsHelper.convertToElastic(e)));
                     }
+                    if (e instanceof ElasticsearchException) {
```

Comment:
Can we serialize the newly added exception and use it here?

Reply:
But then `MissingHistoryOperationsException` needs to extend `ElasticsearchException`, otherwise it cannot be serialized.

Regardless of that I always understood that we want to minimize the number of exceptions that are serializable. This is why I added a header to `ElasticsearchException` instead of creating a new one. Does this still apply? Or do we really want to intoduce a new serializable exception here?

Reply:
> Regardless of that I always understood that we want to minimize the number of exceptions that are serializable. 

If this is the case, let's use the header or metadata here.